pkg/crd/crdvalidations.go

```diff
@@ -0,0 +1,412 @@
+)
+
+var (
+	// Function validation schema properties
+	functionSchemaProps = map[string]apiextensionsv1beta1.JSONSchemaProps{
+		"spec": {
+			Type:        "object",
+			Description: "Specification of the desired behaviour of the Function",
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"environment": environmentReferenceSchema,
+				"package":     functionPackageRefSchema,
+				"secrets":     secretReferenceSchema,
+				"configmaps":  configMapReferenceSchema,
+				"resources": {
+					Type:        "object",
+					Description: "ResourceRequirements describes the compute resource requirements. This is only for newdeploy to set up resource limitation when creating deployment for a function.",
+				},
+				"InvokeStrategy": invokeStrategySchema,
+				"functionTimeout": {
+					Type:        "integer",
+					Description: " FunctionTimeout provides a maximum amount of duration within which a request for a particular function execution should be complete.\nThis is optional. If not specified default value will be taken as 60s",
+				},
+				"idletimeout": {
+					Type:        "integer",
+					Description: "IdleTimeout specifies the length of time that a function is idle before the function pod(s) are eligible for deletion. If no traffic to the function is detected within the idle timeout, the executor will then recycle the function pod(s) to release resources.",
+				},
+			},
+		},
+	}
+
+	// Function validation schema
+	functionSchema = apiextensionsv1beta1.JSONSchemaProps{
+		Type:        "object",
+		Description: "Function in fission is something that Fission executes. It’s usually a module with one entry point, and that entry point is a function with a certain interface.",
+		Properties:  functionSchemaProps,
+	}
+
+	// Function validation object
+	functionValidation = &apiextensionsv1beta1.CustomResourceValidation{
+		OpenAPIV3Schema: &functionSchema,
+	}
+)
+
+var (
+	// Environment validation schema properties
+	environmentSchemaProps = map[string]apiextensionsv1beta1.JSONSchemaProps{
+		"spec": {
+			Type:        "object",
+			Description: "Specification of the desired behaviour of the Environment",
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"version": {
+					Type:        "integer",
+					Description: "Version is the Environment API version",
+				},
+				"runtime": runtimeSchema,
+				"builder": builderSchema,
+				"allowedFunctionsPerContainer": {
+					Type:        "string",
+					Description: "Allowed functions per container. Allowed Values: single, multiple",
+				},
+				"allowAccessToExternalNetwork": {
+					Type:        "boolean",
+					Description: "To enable accessibility of external network for builder/function pod, set to 'true'.",
+				},
+				"resources": {
+					Type:        "object",
+					Description: "The request and limit CPU/MEM resource setting for poolmanager to set up pods in the pre-warm pool.",
```

Comment:
```suggestion
					Description: "The request and limit CPU/MEM resource setting for the pods of the function. Can be overridden at Function in case of newdeployment executor type",
```